plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,945 @@
+                instance of the class and call 'validate_input_yml' on it.
+                If the validation succeeds, this will allow to go next step, 
+                unless this will stop execution based on the fields.
+        """
+        self.log('Validating the Playbook Yaml File..', "INFO")
+        if not self.config:
+            self.status = "success"
+            self.msg = "Configuration is not available in the playbook for validation"
+            self.log(self.msg, "ERROR")
+            return self
+
+        aplist = self.payload.get("config")
+        aplist = self.camel_to_snake_case(aplist)
+        aplist = self.update_site_type_key(aplist)
+        accesspoint_spec = dict(mac_address=dict(required=False, type='str'),
+                    led_brightness_level=dict(required=False, type='int'),
+                    led_status = dict(required=False, type='str'),
+                    location = dict(required=False, type='str'),
+                    ap_name = dict(required=False, type='str'),
+                    ap_name_new = dict(required=False, type='str'),
+                    management_ip_address = dict(required=False, type='str'),
+                    hostname = dict(required=False, type='str'),
+                    device_fields = dict(required=False, type='str'),
+                    ap_selected_field = dict(required=False, type='str'),
+                    )
+        valid_param, invalid_params = validate_list_of_dicts(aplist, accesspoint_spec)
+        if invalid_params:
+            self.msg = "Invalid parameters in playbook: {0}".format(
+                "\n".join(invalid_params)
+            )
+            self.log(self.msg, "ERROR")
+            self.status = "failed"
+            return self
+
+        self.validated_config = valid_param
+        self.msg = "Successfully validated playbook config params:{0}".format(str(valid_param[0]))
+        self.log(self.msg, "INFO")
+        self.status = "success"
+        return self
+
+
+    def get_want(self, ap_config):
+        """
+        Get all Access Point related information from the playbook needed for creation/updation
+         in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            config (dict): A dictionary containing configuration information.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+             Retrieves all Access Point configuration details from the playbook config,
+        excluding any fields not directly related to the Access Point configuration such as
+        'device_fields' and 'ap_selected_field'. The extracted information is stored in the 
+        'want' attribute of the instance for later use in the workflow.
+        """
+        self.log("CHECKIN" + str(ap_config), "INFO")
```

Comment:
Still not using format for string and also log message is not clear.
self.log("AP configuration given in the playbook - '{0}' ".format(str(ap_config)), "INFO")